superset/utils/date_parser.py

```diff
@@ -241,18 +410,20 @@ def get_since_until(  # pylint: disable=too-many-arguments,too-many-locals,too-m
     if time_range and separator in time_range:
         time_range_lookup = [
             (
-                r"^last\s+(day|week|month|quarter|year)$",
-                lambda unit: f"DATEADD(DATETIME('{_relative_start}'), -1, {unit})",
-            ),
-            (
-                r"^last\s+([0-9]+)\s+(second|minute|hour|day|week|month|year)s?$",
-                lambda delta,
-                unit: f"DATEADD(DATETIME('{_relative_start}'), -{int(delta)}, {unit})",  # pylint: disable=line-too-long,useless-suppression
+                r"^(start of|beginning of|end of)\s+(this|last|next|prior)\s+([0-9]+)?\s*(day|week|month|quarter|year)s?$",  # pylint: disable=line-too-long,useless-suppression  # noqa: E501
```

Comment:
It's probably easier to understand the regular expression if you use `re.VERBOSE` mode, something like:

```python
pattern = re.compile(r"""
    ^
    (start\ of|beginning\ of|end\ of)  # Match the phrases 'start of', 'beginning of', or 'end of'
    \s+                                # One or more spaces
    (this|last|next|prior)             # Match time references
    \s+                                # One or more spaces
    ([0-9]+)?                          # Optional number
    \s*                                # Optional whitespace
    (day|week|month|quarter|year)s?    # Match time unit with optional plural 's'
    $
""", re.VERBOSE)
```